src/vmm/src/arch/x86_64/vcpu.rs

```diff
@@ -915,24 +906,37 @@ mod tests {
         // Test configure while using the T2S template.
         let t2a_res = try_configure(&kvm, &mut vcpu, StaticCpuTemplate::T2A);
 
+        let cpu_model = CpuModel::get_cpu_model();
         match &cpuid::common::get_vendor_id_from_host().unwrap() {
             cpuid::VENDOR_ID_INTEL => {
-                assert!(t2_res);
-                assert!(c3_res);
-                assert!(t2s_res);
-                if is_at_least_cascade_lake() {
-                    assert!(t2cl_res);
-                } else {
-                    assert!(!t2cl_res);
-                }
+                assert_eq!(
+                    t2_res,
+                    cpu_model == SKYLAKE_FMS
+                        || cpu_model == CASCADE_LAKE_FMS
+                        || cpu_model == ICE_LAKE_FMS
+                );
+                assert_eq!(
+                    c3_res,
+                    cpu_model == SKYLAKE_FMS
+                        || cpu_model == CASCADE_LAKE_FMS
+                        || cpu_model == ICE_LAKE_FMS
+                );
+                assert_eq!(
+                    t2s_res,
+                    cpu_model == SKYLAKE_FMS || cpu_model == CASCADE_LAKE_FMS
+                );
+                assert_eq!(
+                    t2cl_res,
+                    cpu_model == CASCADE_LAKE_FMS || cpu_model == ICE_LAKE_FMS
+                );
                 assert!(!t2a_res);
             }
             cpuid::VENDOR_ID_AMD => {
                 assert!(!t2_res);
                 assert!(!c3_res);
                 assert!(!t2s_res);
                 assert!(!t2cl_res);
-                assert!(t2a_res);
+                assert_eq!(t2a_res, cpu_model == MILAN_FMS);
```

Comment:
Mhh, I think we don't need the `match` here? As in, this thing could just be

```
assert_eq!(t2_res, [SKYLAKE_FMS, CASCADE_LAKE_FMS, ICE_LAKE_FMS].contains(&cpu_model));
assert_eq!(c3_res, [SKYLAKE_FMS, CASCADE_LAKE_FMS, ICE_LAKE_FMS].contains(&cpu_model));
...
assert_eq!(t2a_res, [MILAN_FMS].contains(&cpu_model));
```

no?